src/main/java/com/spring/app/controllers/MainController.java

```diff
@@ -1,153 +1,126 @@
-        date.setYear(1995);
-
-        Employee employee1 = Employee.builder()
-                .birthday(date)
-                .firstWorkDay(date2)
-                .build();
-        employeeRepository.save(employee1);
-
-        Employee employee2 = Employee.builder()
-                .birthday(date)
-                .firstWorkDay(date2)
-                .build();
-        employeeRepository.save(employee2);
-
-        Vacancy vacancy3 = Vacancy.builder()
-                .company(company1)
-                .dateClosed(date)
-                .dateOpened(date2)
-                .employee(employee1)
-                .name("dgfsg")
-                .build();
-        vacancyRepository.save(vacancy3);
-        vacancy3 = Vacancy.builder()
-                .company(company2)
-                .dateClosed(date)
-                .dateOpened(date2)
-                .employee(employee2)
-                .name("sfgsfds")
-                .build();
-        vacancyRepository.save(vacancy3);
-        vacancy3 = Vacancy.builder()
-                .company(company1)
-                .dateClosed(date)
-                .dateOpened(date2)
-                .employee(employee1)
-                .name("sfhhhds")
-                .build();
-        vacancyRepository.save(vacancy3);
-
-        return "home";
-    }
+        EntityManager session = entityManagerFactory.createEntityManager();
+        try {
+            double age = (Double)session.createNativeQuery("SELECT AVG(vacancy.salary) FROM country INNER JOIN company " +
+                    "ON country.country_id = company.country_id INNER JOIN vacancy ON company.company_id = vacancy.company_id " +
+                    "WHERE country.name = ?1 AND vacancy.opened = TRUE")
+                    .setParameter(1, country)
+                    .getSingleResult();
 
-    @PostMapping("home/findAverageAgeByCountry")
-    public String filter(@RequestParam String country, Map<String, Object> model) {
+            return new ResponseEntity<>(age, HttpStatus.OK);
+        }
+        catch (NoResultException e){
 
-        if (country != null && !country.isEmpty() ) {
-            Country countryModel =  countryRepository.findByName(country).get(0);
 
-            int totalAge = (int) countryModel.getCompanies().stream()
-                    .flatMap(company -> company.getVacancy().stream())
-                    .map(vacancy -> (int)(new Date().getTime() - vacancy.getEmployee().getBirthday().getTime())/ 1000/ 60/ 60/ 24/ 365)
-                    .reduce(0, (x, y) -> x+y);
-            int count = (int) countryModel.getCompanies().stream()
-                    .flatMap(company -> company.getVacancy().stream())
-                    .map(vacancy -> vacancy.getEmployee())
-                    .count();
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+        finally {
+            if(session.isOpen()) session.close();
+        }
+
+    }
+
+    @GetMapping("findAverageAgeByCountry")
```

Comment:
incorrect rest api naming

Reply:
I found this article https://restfulapi.net/resource-naming/... so, do you mean I should use hyphens (-) to improve the readability of URIs??? Like that - "find-average-age-by-country"